tests/backup/backup_restore_basic_test.go

```diff
@@ -1442,3 +1442,220 @@ var _ = Describe("{BackupMultipleNsWithSameLabel}", func() {
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				credMap[provider][cloudCredUID] = credName
+				err := CreateCloudCredential(provider, credName, cloudCredUID, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", credName, orgID, provider))
+				log.InfoD("Created Cloud Credentials with name - %s", credName)
+				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				backupLocationMap[backupLocationUID] = customBackupLocationName
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+			}
+		})
+
+		Step("Register source and destination cluster for backup", func() {
+			log.InfoD("Registering Source and Destination clusters and verifying the status")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			var sem = make(chan struct{}, numberOfSimultaneousBackups)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Taking %d backups", numberOfBackups)
+			for backupLocationUID, backupLocationName := range backupLocationMap {
+				for _, namespace := range bkpNamespaces {
+					for i := 0; i < numberOfBackups; i++ {
+						sem <- struct{}{}
+						time.Sleep(timeBetweenConsecutiveBackups)
+						backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+						backupNames = append(backupNames, backupName)
+						wg.Add(1)
+						go func(backupName string) {
+							defer GinkgoRecover()
+							defer wg.Done()
+							defer func() { <-sem }()
+							err = CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{namespace},
+								labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation: %s", backupName))
+						}(backupName)
+					}
+					wg.Wait()
+				}
+			}
```

Comment:
Continuation of https://github.com/portworx/torpedo/pull/1293#discussion_r1157065367

**You said**: You will not know if it passed or failed until it completes for all namespaces that's something we don't want right?

I don't see how the notification of passing or failing is dependent on _when_ we call `wg.Wait()`. Like, calling `wg.Wait()` outside the main for loop can ensure maximum parallelism. As for whether we know if `CreateBackup` in a goroutine passed/failed, that will occur as soon as `dash.VerifyFatal` is called, right? Because, if it failed, then `GinkgoRecover` will fail the spec immediately (**without waiting for any of the other goroutines, right?**), and if it succeeded, then it's execution as usual.

It's not like we get to know the status of a createbackup _only_ after `wg.Wait()`. As I as far as I can tell, ginkgo will fail the spec as soon as it fails, and not wait for `wg.Wait()` before failing, as there is no interdependency between ginkgo andnumber of currently running goroutines.

Reply:
@mkoppal-px Can you please provide some info here ^

Reply:
Your question is not clear. But the objective is to fail even if one of the backups fail which will happen anyway.
`wg.Wait()` is only for the happy path, where we will ask the main thread to wait till all the go routines are done.

Reply:
Since `wg.Wait()` is only for the happy path, we can put it out of all the loops, right? (for max concurrency)

Reply:
Okay will do it.